api/app/wildfire_one/schema_parsers.py

```diff
@@ -240,3 +242,51 @@ def parse_hourly_actual(station_code: int, hourly):
     # don't write the HourlyActual to our database if every value is invalid. If even one
     # weather variable observed is valid, write the HourlyActual to DB.
     return None if is_obs_invalid else hourly_actual
+
+
+async def weather_station_group_mapper(raw_station_groups_by_owner: Generator[dict, None, None]) -> List[WeatherStationGroupsByOwner]:
```

Comment:
This function feels really inefficient to me and has a lot of changes to the structure of the data.
1. A generator to a flat list of objects.
2. Sorting that flat list.
3. Grouping the flat list
4. Group to dictionary
5. Dictionary to list of final objects to be returned to the front-end

I'm sure there is a better way, but this was kind of proof of concept. I guess this won't be exercised that often.